### DISCUSSION TITLE
but the files aren't 4.10kb

### DISCUSSION THREAD

**Discussion by hoodie** [Byron/dua-cli#121] [Q&A] [answered]
Hi there,
and thank you for building this tool, I love it!

Except...

Please tell me if I misunderstand the UI, but `dua` shows most small files as being 4.10kb in size.
You can even see this in the asciinema recording. Is that intentional?

Thank you

**Comment by Byron** [comment, marked as answer]
I think what you are looking for is `dua -A` for the apparent size. By default, `dua` counts the actual size on disk which is a multiple of the block size, in this case it's 4kb for one block. Small files consume at least one block.

Doing that is similar to `du`'s default, I think.

Hope that clarifies what's going on here.

**Reply by polarathene** [reply]
> Doing that is similar to `du`'s default, I think.

I was actually confused when I used `du -sx --bytes` (_optionally followed by `--si` / `-h`_) as the size reported was quite a bit larger than DUA was reporting.

Turned out this was due to `du` counting the 4KiB per directory, where `/usr/share` had quite a lot of nested directories that were effectively empty. DUA ignored all that, while `du` counted it (_and it doesn't seem to offer any option to opt-out_). On most file systems I assume the min block size of 4096 bytes would still apply though?

Other tools seem to also exclude that and treat it as empty instead of several MB. Likewise I don't think DUA has an option to count that if it does actually contribute to disk usage.

Probably rarely relevant, but I was doing some analysis for optimizing the size of a docker image (_using `dnf --installroot`_) and this disparity between the two tools tripped me up a bit on which to trust and why they differed 😅 

---

**UPDATE:**
- `du` 9.5 (2024) on Fedora 41 was correctly using `--apparent-size` to treat empty directories as `0` bytes used.
- `du` 8.32 (2020) on OpenSUSE Leap 15.6 (BCI Base latest OCI image) however reports `4096` bytes for an empty directory. Presumably a bug with `--apparent-size` that got fixed.

**Reply by Byron** [reply]
Thanks for sharing!
It would probably be more correct if `dua` would also count empty directories as consumer of an inode, which typically clocks in at 4KB. Please feel free to raise an issue about this, or submit a PR with a fix.
